mono/metadata/threads.c

```diff
@@ -749,6 +749,20 @@ typedef struct {
 	MonoCoopSem registered;
 } StartInfo;
 
+static void
+fire_attach_profiler_events (gsize tid)
+{
+	MONO_PROFILER_RAISE (thread_started, (tid));
+
+	MonoThreadInfo *info = mono_thread_info_current ();
+	MONO_PROFILER_RAISE (root_register, (
+		info->stack_start_limit,
+		(char*)info->stack_end - (char*)info->stack_start_limit,
+		MONO_ROOT_SOURCE_STACK,
+		MONO_UINT_TO_NATIVE_THREAD_ID (tid),
```

Comment:
This cast looks wrong. In the `thread_started` event above, we pass the result of `MONO_NATIVE_THREAD_ID_TO_UINT (tid)` (see the call site of this function), but here, we convert it back to native thread ID before passing it.